core/src/main/scala/org/apache/spark/sql/extensions/rules.scala

```diff
@@ -50,51 +76,162 @@ case class TiResolutionRule(getOrCreateTiContext: SparkSession => TiContext)(
       val tiDBRelation =
         TiDBRelation(tiSession, TiTableReference(dbName, tableName, sizeInBytes), meta)(
           sqlContext)
-      // Use SubqueryAlias so that projects and joins can correctly resolve
-      // UnresolvedAttributes in JoinConditions, Projects, Filters, etc.
-      newSubqueryAlias(tableName, LogicalRelation(tiDBRelation))
+      if (withSubQueryAlias) {
+        // Use SubqueryAlias so that projects and joins can correctly resolve
+        // UnresolvedAttributes in JoinConditions, Projects, Filters, etc.
+        SubqueryAlias(tableName, LogicalRelation(tiDBRelation))
+      } else {
+        LogicalRelation(tiDBRelation)
+      }
     }
 
   override def apply(plan: LogicalPlan): LogicalPlan =
     plan transformUp resolveTiDBRelations
 
   protected def resolveTiDBRelations: PartialFunction[LogicalPlan, LogicalPlan] = {
-    case i @ InsertIntoTable(UnresolvedRelation(tableIdentifier), _, _, _, _)
+    case i @ InsertIntoStatement(UnresolvedRelation(tableIdentifier), _, _, _, _)
         if tiCatalog
-          .catalogOf(tableIdentifier.database)
+          .catalogOf(if (tableIdentifier.size == 1) None else Some(tableIdentifier.head))
```

Comment:
we can move this logic to a new function `catalogOf(Seq[xxx])`

Reply:
done